### ISSUE TITLE
nnUNet spawns way to many processes

### ISSUE THREAD

**Issue by Chris-N-K** [MIC-DKFZ/nnUNet#1022]
Issue:
nnUNet spawns way to many processes during training. 
During training (3d_fullres) a lot of very low CPU usage (0-2%) processes are spawned. All cores are in use.
The environment variable: nnUNet_n_proc_DA has no influence.
Epoch training times in the Task02 3d_fullres benchmark are around 240 seconds and at 170 second for 2d.
GPU usage and power draw are maxed out.

I'm pretty sure nnUNet should not spawn this many processes. I'm not sure but I guess the training times are somewhat lengthy as well.

Environment:
GPU: Tesla T4
Driver Version: 460.91.03    CUDA Version: 11.2    cuDNN Version: 8.1.1

CPU: Intel(R) Xeon(R) Gold 5218R CPU @ 2.10GHz

Operating System: Debian 10 Buster

pyTorch: 1.11.0 py3.9_cuda11.3_cudnn8.2.0_0
nnUNet: 1.7.0 pypi_0

**Comment by FabianIsensee** [comment]
Hey so during training you get `nnUNet_n_proc_DA` data augmentation workers for the training data, another `nnUNet_n_proc_DA // 2` workers for the validation data. Then you have the main process. Finally during validation you will have `nnUNet_def_n_proc` (default=8) segmentation export workers. That is all. Yes this number is large in total but only some of these are active at a time so it's all fine

Have you tried `export OMP_NUM_THREADS=1`?

**Comment by ch-n** [comment]
Hi @FabianIsensee thanks for the reply.
At the moment I seem to face an additional / other issue were nnUNet still spawns to many processes but they do all show moderate to high cpu usage (30 - 70 %). That might be of a custom loss function and pytorch issues I'm not sure in this regard.

I will try your suggestion, maybe it will have an effect on this phenomenon as well.

Best regards
Chris

**Comment by FabianIsensee** [comment]
Hey, it doesn't really matter what each individual process uses. What is the overall usage of your CPU? Is it pegged? Or does it have some headroom?

**Comment by ch-n** [comment]
Every available core is used. That are 80 cores in total, but some might be used by other processes as well.

**Comment by FabianIsensee** [comment]
This looks like a CPU bottleneck to me. You may need to ask the others to be nicer on the system ;-)

**Comment by ch-n** [comment]
I am the sole user of the system during training ^^'

**Comment by FabianIsensee** [comment]
In that case let the other processes pause for a while ;-)
If you have 80 cores then one standard nnU-Net training should not bottleneck!

**Comment by Chris-N-K** [comment]
No problem any more with version 2
